### PR TITLE
Implement post instanciation method call on service

### PR DESCRIPTION
**Pull request by Spy-Seth**
- Add check cyclic dependencies on call hook (not needed)
- [x] Rename the `ServiceArgumentCollection` to `ArgumentCollection` and `ServiceArgument` to `Argument`

### CODE REVIEW

File: src/Call.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    this.serviceArgumentCollection = serviceArgumentCollection;
+};
+
+/**
+ * @param {Container} container
+ * @param {Object} instance
+ */
+Call.prototype.trigger = function trigger(container, instance) {
+    if (!instance[this.methodName]) {
+        throw new Error('Can\'t call the given method: "' + this.methodName + '" does not exist on the given instance.');
```
Comment (shouze) [inline src/Call.js]: So... we will never know until runtime that the method is unknown isn't it? So... the big question is... do we have to be able to launch some kind of checks at container's instanciation? In this case we maybe should collect all checks in a heap and delegate the exception throwing to a dedicated class that will be specialized in heap checks assertions?

Comment (Spy-Seth) [inline src/Call.js]: I've add check on container initialization. These check stay "just in case".

File: src/Container.js [hunk 2]
```diff
@@ -15,6 +15,11 @@ var Container = function Container(serviceDefinitionCollection, parameterCollect
     this.parameterCollection = parameterCollection;
 
     this.serviceStorage = new ServiceStorage();
+
+    if (process.env.NODE_ENV === 'development') {
```
Comment (shouze) [inline src/Container.js]: Maybe, instead of being hard coupled to `NODE_ENV` here... this could become an argument of the Container constructor?

Comment (shouze) [inline src/Container.js]: Something like:
```javascript
function Container(serviceDefinitionCollection, parameterCollection, boolean: validateDefinitions)
```

Comment (Spy-Seth) [inline src/Container.js]: Ho :scream: That's a really good idea!

Comment (Spy-Seth) [inline src/Container.js]: I will need to add it to the `ContainerFactory` too.

**Comment by shouze** [comment]
LGTM, good to squash then merge! :+1:

**Pull request merged**